src/GitVersionCore/Helpers/EnvironmentHelper.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace GitVersion.Helpers
+{
+    public class EnvironmentHelper
+    {
+        public static string GetEnvironmentVariableForProcess(string envVar)
+        {
+            return Environment.GetEnvironmentVariable(envVar, EnvironmentVariableTarget.Process);
```

Comment:
Could we make an abstraction on top of `Environment` here so it's translated to something less specific like `IDictionary<string, string>`? That would make testing easier and the code more portable.

Reply:
I am not sure exactly as to how could I improve here and provide a useful abstraction.  Seems like Mono https://github.com/mono/mono/blob/master/mcs/class/corlib/System/Environment.cs uses the same approach as we have. In case there are multiple env vars, a dictionary would make sense but in this case we seek the value of a single env var and this should be fine

Reply:
I just want a translation so our "core" isn't dependent on the `System.Environment` class itself. Perhaps `Func<string, string>` is a better abstraction?

Reply:
Using `Func<string, string>` has the advantage that one can use `System.Environment.GetEnvironmentVariable` (or even the currently used `Helpers.EnvironmentHelper.GetEnvironmentVariableForProcess`) directly at the call site.

What is your preference here? Turning this formatter into a class and use ctor injection or rather give the currently used string extension method an additional parameter?
(e.g. `public static string FormatWith<T>(this string template, T source, Func<string, string> getEnvironmentVariable)`)

Reply:
`GitVersionCore\Authentication.cs`,  `GitVersionCore\BuildServers\*.cs` etc all use a naked call to `System.Environment.GetEnvironmentVariable`. How exactly does my patch create a problem when there are already existing reference to System.Environment in the GitVersionCore? Maybe abstracting all the System.Environment references in the GitVersionCore can be a part of a cleanup but I am not sure how drastically does it change things here.

Reply:
True, I did not think of the existing calls to `System.Environment`. I agree; let's address that abstraction in a future PR.